etc/kayobe/ansible/rekey-hosts.yml

```diff
@@ -0,0 +1,97 @@
+---
+# Playbook to rotate SSH keys across the cloud. By default it will rotate the
+# standard keys used by kayobe/kolla-ansible, but it can be configured for any
+# keys.
+
+- name: Rekey hosts
+  hosts: overcloud,seed,seed-hypervisor,infra-vms
+  gather_facts: false
+  vars:
+    ansible_ssh_common_args: "-o StrictHostKeyChecking=no"
+    existing_private_key_path: "{{ ssh_private_key_path }}"
+    existing_public_key_path: "{{ ssh_public_key_path }}"
+    new_private_key_path: "{{ ssh_private_key_path }}"
+    new_public_key_path: "{{ ssh_public_key_path }}"
+    new_key_type: "{{ ssh_key_type }}"
+    rekey_users:
+      - stack
+      - kolla
+    rekey_remove_existing_key: false
+  tasks:
+    - name: Stat existing private key file
+      ansible.builtin.stat:
+        path: "{{ existing_private_key_path }}"
+      register: stat_result
+      delegate_to: localhost
+      run_once: true
+
+    - name: Fail when existing private key does not exist
+      ansible.builtin.fail:
+        msg: "No existing private key file found. Check existing_private_key_path is set correctly."
+      when:
+        - not stat_result.stat.exists
+      delegate_to: localhost
+      run_once: true
+
+    - name: Stat existing public key file
+      ansible.builtin.stat:
+        path: "{{ existing_public_key_path }}"
+      register: stat_result
+      delegate_to: localhost
+      run_once: true
+
+    - name: Fail when existing public key does not exist
+      ansible.builtin.fail:
+        msg: "No existing public key file found. Check existing_public_key_path is set correctly."
+      when:
+        - not stat_result.stat.exists
+      delegate_to: localhost
+      run_once: true
+
+    - name: Generate a new SSH key
+      community.crypto.openssh_keypair:
+        path: "{{ existing_private_key_path }}_new"
```

Comment:
Should this use new_private_key_path?

Reply:
The key gets moved to `"{{ new_private_key_path }}"` later on. the only criteria for the path in that line is that it can't conflict with the existing key.

If it's purely `path: "{{ new_private_key_path }}"` then the new path cannot equal the existing path. 

`"{{ new_private_key_path }}_new"` would work in almost every scenario, except when 
`"{{ new_private_key_path }}_new" == "{{ existing_private_key_path }}"` (which is an incredibly niche situation but still)

Reply:
The main thing I was thinking is if you change the key type then the temporary name would look wrong

Reply:
I see what you mean, however the user shouldn't ever see that temporary name, since it's immediately removed